### DISCUSSION TITLE
Add the picked image to a XR running time image library

### DISCUSSION THREAD

**Discussion by Isaac-dot** [yasirkula/UnityNativeGallery#257] [General]
i would like to add the picked image to XR running time image library in running time, and the XR running time image library is created in running time as well. following is my code. there is no error. it just doesnot work.
 `using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.XR;
using UnityEngine.XR.ARFoundation;
using UnityEngine.XR.ARSubsystems;
using UnityEngine.UI;

public class ImagePicker : MonoBehaviour
{

    
    private ARTrackedImageManager trackedImageManager;

    private void Awake()
    {
        trackedImageManager = FindObjectOfType<ARTrackedImageManager>();
    }

    

    // Start is called before the first frame update
    public void PickImage()
    {
        NativeGallery.Permission permission = NativeGallery.GetImageFromGallery( ( path ) =>
        {
            Texture2D PickedImage = NativeGallery.LoadImageAtPath( path );
            Debug.Log( "Image path: " + path );
            var runtimeLibrary = trackedImageManager.CreateRuntimeLibrary();
        if (runtimeLibrary is MutableRuntimeReferenceImageLibrary mutableLibrary)
        {
            if( path != null )
            {
                mutableLibrary.ScheduleAddImageJob(
                    duplicateTexture(PickedImage),
                    "my new image",
                    0.5f /* 50 cm */);
            }
        }           

        } );

        Debug.Log( "Permission result: " + permission );
    }
    Texture2D duplicateTexture(Texture2D source)
    {
        RenderTexture renderTex = RenderTexture.GetTemporary(
                    source.width,
                    source.height,
                    0,
                    RenderTextureFormat.Default,
                    RenderTextureReadWrite.Linear);

        Graphics.Blit(source, renderTex);
        RenderTexture previous = RenderTexture.active;
        RenderTexture.active = renderTex;
        Texture2D readableText = new Texture2D(source.width, source.height);
        readableText.ReadPixels(new Rect(0, 0, renderTex.width, renderTex.height), 0, 0);
        readableText.Apply();
        RenderTexture.active = previous;
        RenderTexture.ReleaseTemporary(renderTex);
        return readableText;
    }
}
`

**Comment by yasirkula** [comment]
2 things I can recommend are:

- Call if( path != null ) at the top of LoadImageAtPath because otherwise LoadImageAtPath will throw an exception for null paths
- Set LoadImageAtPath's markTextureNonReadable parameter to false so that you won't need to call the duplicateTexture function at all :D

Though, these are probably not the cause of the problem. I don't know what image library is but I'd recommend putting Debug.Logs to random parts of your code and checking logcat to see if they're called, as well as to see if there are any error messages.
